### PR TITLE
Avijit/cpp app

### PR DESCRIPTION
**Pull request by avijit-nervana**
* This PR creates a C++ example of using a pre-trained network to classify images (based on TensorFlow label_image example)
* The single model example (`infer_single`) will be extended to support multiple inference threads in a separate PR.
* This PR also creates a placeholder for a C++ inference example application with multiple inference models.

### CODE REVIEW

File: examples/cpp/infer_multi/infer_multi.cpp [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+// the computation
+tf::Status SetNGraphBackend(const string& backend_name) {
+  // Select a backend
+  tf::Status status =
+      tf::ngraph_bridge::BackendManager::SetBackendName(backend_name);
+  return status;
+}
+
+// Create a simple computation graph and run
+void RunSimpleNetworkExample() {
```
Comment (shresthamalik) [inline examples/cpp/infer_multi/infer_multi.cpp]: May be we can remove this example. It is in examples/tf_cpp_examples/hello_tf.cpp

File: examples/cpp/infer_multi/infer_multi.cpp [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+  // Run the MatMul example
+  auto session = CreateSession("inception_v3_2016_08_28_frozen.pb");
+
+  // Get the image from disk as a float array of numbers, resized and normalized
+  // to the specifications the main graph expects.
+  std::vector<tf::Tensor> resized_tensors;
+  tf::Status read_tensor_status = ReadTensorFromImageFile(
+      "grace_hopper.jpg", 299 /*input_height*/, 299 /*input_width*/,
+      0.0 /*input_mean*/, 255 /*input_std*/, &resized_tensors);
```
Comment (shresthamalik) [inline examples/cpp/infer_multi/infer_multi.cpp]: Should we park the image in the repo or add instructions how to get the image and frozen graph?

File: examples/cpp/infer_single/infer.cpp [hunk 3]
```diff
@@ -0,0 +1,266 @@
+/*******************************************************************************
```
Comment (shresthamalik) [inline examples/cpp/infer_single/infer.cpp]: Please change the name of the folder and file, infer_single_<what>?

File: examples/cpp/infer_multi/infer_multi.cpp (hunk 2 above)
Comment (avijit-nervana) [reply examples/cpp/infer_multi/infer_multi.cpp]: Will add instructions.

File: examples/cpp/infer_multi/infer_multi.cpp (hunk 1 above)
Comment (avijit-nervana) [reply examples/cpp/infer_multi/infer_multi.cpp]: I guess so. Will do.

File: examples/cpp/infer_single/infer.cpp (hunk 3 above)
Comment (avijit-nervana) [reply examples/cpp/infer_single/infer.cpp]: Done.

File: examples/cpp/infer_multi/infer_multi.cpp [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+    cout << "Backend: " << backend_name << std::endl;
+  }
+}
+
+// Sets the specified backend. This backend must be set BEFORE running
+// the computation
+tf::Status SetNGraphBackend(const string& backend_name) {
+  // Select a backend
+  tf::Status status =
+      tf::ngraph_bridge::BackendManager::SetBackendName(backend_name);
```
[the remark this answers is not in the document]
Comment (avijit-nervana) [reply examples/cpp/infer_multi/infer_multi.cpp]: For now we removed building of infer_multi.

File: examples/cpp/infer_multi/infer_multi.cpp [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+                                          std::vector<tf::Tensor>* out_tensors);
+
+extern tf::Status PrintTopLabels(const std::vector<tf::Tensor>& outputs,
+                                 const string& labels_file_name);
+
+// Prints the available backends
+void PrintAvailableBackends() {
+  // Get the list of backends
+  auto supported_backends =
+      tf::ngraph_bridge::BackendManager::GetSupportedBackendNames();
```
[the remark this answers is not in the document]
Comment (avijit-nervana) [reply examples/cpp/infer_multi/infer_multi.cpp]: We will need to revisit the config APIs - so I am deliberately not using them here.

File: examples/cpp/infer_multi/infer_multi.cpp [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+  // Until we get a library fully running
+  if (tf::ngraph_bridge::ngraph_tf_is_grappler_enabled()) {
+    options.config.mutable_graph_options()
+        ->mutable_rewrite_options()
+        ->add_custom_optimizers()
+        ->set_name("ngraph-optimizer");
+
+    options.config.mutable_graph_options()
+        ->mutable_rewrite_options()
+        ->set_min_graph_nodes(-1);
```
[the remark this answers is not in the document]
Comment (avijit-nervana) [reply examples/cpp/infer_multi/infer_multi.cpp]: For now - we're not going to build the infer_multi. Will remove this.

File: examples/cpp/infer_multi/infer_multi.cpp [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
```
[the remark this answers is not in the document]
Comment (avijit-nervana) [reply examples/cpp/infer_multi/infer_multi.cpp]: Project multi removed for now.

**Pull request merged**